app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -52,8 +84,92 @@ export default class AMRHeatmapVis extends React.Component {
     return [sampleLabels, geneLabels, alleleLabels];
   }
 
+  //*** Callback functions for the heatmap ***
+
+  colorFilter = (value, node, originalColor, _, colorNoValue) => {
+    // Leave zero values grey
+    return value > 0 ? originalColor : colorNoValue;
+  };
+
+  onNodeHover = node => {
+    this.setState({ nodeHoverInfo: this.getTooltipData(node) });
+  };
+
+  onNodeHoverMove = (_, event) => {
+    if (event) {
+      this.setState({
+        tooltipLocation: {
+          left: event.pageX,
+          top: event.pageY,
+        },
+      });
+    }
+  };
+
+  onNodeHoverOut = () => {
+    this.setState({ nodeHoverInfo: null });
+  };
+
   //*** Following functions must be called after the component has updated ***
-  //*** (i.e. after the component has requested AMR data and updated state) ***
+
+  getTooltipData(node) {
+    const { samplesWithAMRCounts, selectedOptions } = this.props;
+    const {
+      sampleLabels,
+      geneLabels,
+      alleleLabels,
+      alleleToGeneMap,
+    } = this.state;
+    const sampleName = sampleLabels[node.columnIndex].label;
+    let amrCountIdentifier = "";
+    let gene = "";
+    let allele = "";
+    switch (selectedOptions.viewLevel) {
+      case VIEW_LEVEL_ALLELES: {
+        amrCountIdentifier = alleleLabels[node.rowIndex].label;
+        allele = amrCountIdentifier;
+        break;
+      }
+      case VIEW_LEVEL_GENES: {
+        amrCountIdentifier = geneLabels[node.rowIndex].label;
+        gene = amrCountIdentifier;
+        break;
+      }
+      default: {
```

Comment:
looks like you don't need the `default break`

Reply:
It's good practice though, no?